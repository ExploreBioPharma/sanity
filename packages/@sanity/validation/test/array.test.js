const {Rule} = require('../src')

describe('array', () => {
  test('required constraint', () => {
    const rule = Rule.array().required()
    expect(rule.validate(null)).toMatchSnapshot('required: null')
    expect(rule.validate(undefined)).toMatchSnapshot('required: undefined')
    expect(rule.validate([])).toMatchSnapshot('required: empty array')
    expect(rule.validate(['hei'])).toMatchSnapshot('required: valid')
  })

  test('min length constraint', () => {
    const rule = Rule.array().min(2)
    expect(rule.validate(['a'])).toMatchSnapshot('min length: too short')
    expect(rule.validate(['a', 'b', 'c'])).toMatchSnapshot('min length: valid')
  })

  test('max length constraint', () => {
    const rule = Rule.array().max(2)
    expect(rule.validate(['a', 'b', 'c', 'd'])).toMatchSnapshot('max length: too long')
    expect(rule.validate(['a'])).toMatchSnapshot('max length: valid')
  })

  test('exact length constraint', () => {
    const rule = Rule.array().length(2)
    expect(rule.validate(['a', 'b', 'c'])).toMatchSnapshot('exact length: too long')
    expect(rule.validate(['a'])).toMatchSnapshot('exact length: too short')
    expect(rule.validate(['a', 'b'])).toMatchSnapshot('exact length: valid')
  })

  test.skip('unique constraint (default, simple values)', () => {
    const rule = Rule.array().unique()
    expect(rule.validate(['a', 'b'])).toMatchSnapshot('unique: valid')
    expect(rule.validate(['a', 'a'])).toMatchSnapshot('unique: duplicates')
  })
})
