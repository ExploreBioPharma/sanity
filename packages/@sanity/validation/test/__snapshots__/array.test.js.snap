// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array unique: duplicates 1`] = `
Array [
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      0,
    ],
  },
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      1,
    ],
  },
]
`;

exports[`array unique: valid 1`] = `Array []`;

exports[`boolean unique: duplicates 1`] = `
Array [
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      0,
    ],
  },
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      2,
    ],
  },
]
`;

exports[`boolean unique: valid 1`] = `Array []`;

exports[`exact length: too long 1`] = `
Array [
  Object {
    "item": [ValidationError: Array must have exactly 2 items],
    "level": "error",
  },
]
`;

exports[`exact length: too short 1`] = `
Array [
  Object {
    "item": [ValidationError: Array must have exactly 2 items],
    "level": "error",
  },
]
`;

exports[`exact length: valid 1`] = `Array []`;

exports[`max length: too long 1`] = `
Array [
  Object {
    "item": [ValidationError: Array must have at most 2 items],
    "level": "error",
  },
]
`;

exports[`max length: valid 1`] = `Array []`;

exports[`min length: too short 1`] = `
Array [
  Object {
    "item": [ValidationError: Array must have at least 2 items],
    "level": "error",
  },
]
`;

exports[`min length: valid 1`] = `Array []`;

exports[`numeric unique: duplicates 1`] = `
Array [
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      0,
    ],
  },
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      2,
    ],
  },
]
`;

exports[`numeric unique: valid 1`] = `Array []`;

exports[`object unique: duplicates 1`] = `
Array [
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      0,
    ],
  },
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      3,
    ],
  },
]
`;

exports[`object unique: valid 1`] = `Array []`;

exports[`required: empty array 1`] = `Array []`;

exports[`required: null 1`] = `
Array [
  Object {
    "item": [ValidationError: Expected type "Array", got "null"],
    "level": "error",
  },
  Object {
    "item": [ValidationError: This field is required],
    "level": "error",
  },
]
`;

exports[`required: undefined 1`] = `
Array [
  Object {
    "item": [ValidationError: This field is required],
    "level": "error",
  },
]
`;

exports[`required: valid 1`] = `Array []`;

exports[`simple unique: duplicates 1`] = `
Array [
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      0,
    ],
  },
  Object {
    "item": [ValidationError: Array cannot contain duplicate values],
    "level": "error",
    "path": Array [
      3,
    ],
  },
]
`;

exports[`simple unique: valid 1`] = `Array []`;
